java/src/main/java/com/codurance/training/tasks/actions/TaskActionHandlerImpl.java

```diff
@@ -0,0 +1,94 @@
+package com.codurance.training.tasks.actions;
+
+import com.codurance.training.tasks.Task;
+
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TaskActionHandlerImpl implements TaskActionHandler {
+
+    private final PrintWriter out = new PrintWriter(System.out);
+
+    private static long lastId = 0;
+
+    private static final Map<String, List<Task>> tasks = new LinkedHashMap<>();
+
+    public void show() {
+        for (Map.Entry<String, List<Task>> project : tasks.entrySet()) {
+            out.println(project.getKey());
+            for (Task task : project.getValue()) {
+                out.printf("    [%c] %d: %s%n", (task.isDone() ? 'x' : ' '), task.getId(), task.getDescription());
+            }
+            out.println();
+        }
+    }
+
+    public void add(String commandLine) {
+        String[] subcommandRest = commandLine.split(" ", 2);
+        String subcommand = subcommandRest[0];
+        if (subcommand.equals("project")) {
+            addProject(subcommandRest[1]);
+        } else if (subcommand.equals("task")) {
+            String[] projectTask = subcommandRest[1].split(" ", 2);
+            addTask(projectTask[0], projectTask[1]);
+        }
+    }
+
+    public void addProject(String name) {
+        tasks.put(name, new ArrayList<Task>());
+    }
+
+    private void addTask(String project, String description) {
+        List<Task> projectTasks = tasks.get(project);
+        if (projectTasks == null) {
+            out.printf("Could not find a project with the name \"%s\".", project);
+            out.println();
+            return;
+        }
+        projectTasks.add(new Task(nextId(), description, false));
+    }
+
+    public void check(String idString) {
+        setDone(idString, true);
+    }
+
+    public void uncheck(String idString) {
+        setDone(idString, false);
+    }
+
+    public void setDone(String idString, boolean done) {
+        int id = Integer.parseInt(idString);
+        for (Map.Entry<String, List<Task>> project : tasks.entrySet()) {
+            for (Task task : project.getValue()) {
+                if (task.getId() == id) {
+                    task.setDone(done);
+                    return;
+                }
+            }
+        }
+        out.printf("Could not find a task with an ID of %d.", id);
+        out.println();
+    }
+
+    public void help() {
+        out.println("Commands:");
+        out.println("  show");
+        out.println("  add project <project name>");
+        out.println("  add task <project name> <task description>");
+        out.println("  check <task ID>");
+        out.println("  uncheck <task ID>");
+        out.println();
+    }
+
+    public void error(String command) {
+        out.printf("I don't know what the command \"%s\" is.", command);
+        out.println();
+    }
+
+    public static long nextId() {
+        return ++lastId;
+    }
```

Comment:
class needs to be segregated as Single Responsibility is not achieved

Reply:
Done